File: 2024-2/0684/unidad06.d_Web_04/SoftPub/SoftPubGrupoBussiness/LineaBo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubGrupoControler.DAO;
using SoftPubGrupoControler.DAOImpl;
using SoftPubGrupoModel;

namespace SoftPubGrupoBussiness
{
    public class LineaBo
    {
        private LineaDAO lineaDAO;

        public LineaBo()
        {
            this.lineaDAO = new LineaDAOImpl();
        }

        public BindingList<Object> listarTodos()
        {
            return this.lineaDAO.listarTodos();
        }

        public BindingList<Object> listarPorGrupo(int? idGrupo)
        {
            return this.lineaDAO.listarPorGrupo(idGrupo);
        }

        public Linea obtenerPorId(int idGrupo, int idLinea)
        {
            return this.lineaDAO.obtenerPorId(idGrupo, idLinea);
        }
    }
}

File: 2024-2/Proyecto DAO SoftPub Java/SoftPubGrupoBO/src/pe/edu/pucp/softpub/grupo/bo/LineaBO.java
package pe.edu.pucp.softpub.grupo.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpub.grupo.dao.LineaDAO;
import pe.edu.pucp.softpub.grupo.daoImpl.LineaDAOImpl;
import pe.edu.pucp.softpub.grupo.model.Grupo;
import pe.edu.pucp.softpub.grupo.model.Linea;

public class LineaBO {
    
    private LineaDAO lineaDAO;
    
    public LineaBO(){
        this.lineaDAO = new LineaDAOImpl();
    }
 
    public Integer insertar(Integer idGrupo, Integer idLinea, String nombre){
        Linea linea = this.crearLinea(idGrupo, idLinea, nombre);
        return this.lineaDAO.insertar(linea);
    }
    
    public Integer modificar(Integer idGrupo, Integer idLinea, String nombre){
        Linea linea = this.crearLinea(idGrupo, idLinea, nombre);
        return this.lineaDAO.modificar(linea);
    }
    
    public Integer eliminar(Integer idGrupo, Integer idLinea){
        Linea linea = this.crearLinea(idGrupo, idLinea, null);
        return this.lineaDAO.eliminar(linea);
    }
    
    public ArrayList<Linea> listarTodos(Integer idGrupo){
        return this.lineaDAO.listarTodos(idGrupo);
    }
    
    public Linea obtenerPorId(Integer idGrupo, Integer idLinea){
        Linea linea = this.crearLinea(idGrupo, idLinea, null);
        return this.lineaDAO.obtenerPorId(linea);
    }
    
    private Linea crearLinea(Integer idGrupo, Integer idLinea, String nombre){
        Grupo grupo = new Grupo();
        grupo.setIdGrupo(idGrupo);
        
        Linea linea = new Linea();
        linea.setGrupo(grupo);
        linea.setIdLinea(idLinea);
        linea.setNombre(nombre);
        return linea;
    }
    
}
